Fix #69 Adapt URI

cdamus opened the pull request:
- make adaptUri() method more robust (e.g. for spaces in path)
- also fix #126 command deserialization with absolute resource URI
- remove decode overload which expected RS and Model URI
- refactor ModelRepository#addModel to not only get resource, but actually also create it (fixes #124)
- Rename and refactor `readResource` to `readPayload` (fixes #58 )

edgarmueller committed:
Remove decode overload which expected RS and Model URI

Refactor ModelRepository#addModel to not only get resource, but
actually also create it

cdamus committed:
Fix #69 Adapt URI

- make adaptUri() method more robust (e.g. for spaces in path)
- also fix #126 command deserialization with absolute resource URI

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

edgarmueller commented on com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/DefaultJsonCodec.java:
```diff
@@ -53,7 +48,30 @@ public JsonNode encode(EObject obj) throws EncodingException {
+	@Override
+	public Optional<EObject> decode(URI workspaceURI, String payload) throws DecodingException {
+		URI uri = URI.createURI("virtual.json").resolve(workspaceURI);
+		final JsonResource jsonResource = new JsonResource(uri, getObjectMapper());
```
The body is the same as for the other overload, could we just extract a method here or unify both methods?

edgarmueller commented on com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/XmiCodec.java:
```diff
@@ -53,7 +53,14 @@ public JsonNode encode(EObject eObject) throws EncodingException {
+	}
+	
+	@Override
+	public Optional<EObject> decode(URI workspaceURI, String payload) throws DecodingException {
```
We could unify with this method with the one from above.

edgarmueller reviewed (approved):
Looks good to me!

cdamus committed:
Address code review comments #128

- common implementation of decode overloads
- fix parameter ordering in the overload
- throw exception instead of hiding exception in
  the resource that the client will never see (which
  then also aligns with the XmiCodec)
- pull in fix for local resource creation in JUnit test
   from PR #132

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

ndoschek committed:
Merge branch 'master' into fix/69-adapt-uri

ndoschek commented:
Summary of fixed issues with this PR:
Fixes #58 
Fixes #69 
Fixes #95 
Fixes #124 
Fixes #126
